rust/lance-linalg/src/kmeans.rs

```diff
@@ -179,6 +180,15 @@ impl<T: ArrowFloatType + Dot + Cosine + L2> KMeanMembership<T> {
             }
         });
 
+        if empty_count as f32 / self.k as f32 > 0.5 {
+            // TODO: describe what user should do in response to this message.
+            // This seems to be very common in PQ.
+            warn!(
+                "KMeans: more than half of clusters are empty: {} of {}",
+                empty_count, self.k
+            );
+        }
```

Comment:
@eddyxu I could use your opinion here. We hit this a lot in our tests as part of the PQ training. I see:

```
[2023-11-14T18:20:32Z WARN  lance_linalg::kmeans] KMeans: more than half of clusters are empty: 255 of 256
```

Is there something a user should do in response to this? the number of clusters is controlled by `num_bits`, but Lance errors if you change it to anything but 8 so it is stuck at `2^8 = 256` clusters. It's annoying to have a warning that doesn't give any guidance on how to fix it.

At the very least, the changes as is will make it less noisy.